pkg/build/api/validation/validation.go

```diff
@@ -126,6 +150,60 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) field.ErrorList {
 	return allErrs
 }
 
+// ValidateBuildConfigStatusUpdate validates a status update for a BuildConfig
+func ValidateBuildConfigStatusUpdate(config *buildapi.BuildConfig, older *buildapi.BuildConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(config.Spec.Triggers) != len(older.Spec.Triggers) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec.triggers"), config.Spec.Triggers, "cannot change triggers in status update"))
+		return allErrs
+	}
+	for i := range config.Spec.Triggers {
+		allErrs = append(allErrs, validateBuildConfigTriggerStatusUpdate(config.Spec.Triggers[i], older.Spec.Triggers[i], field.NewPath("spec.triggers").Index(i))...)
+	}
+	return allErrs
+}
+
+func validateBuildConfigTriggerStatusUpdate(trigger buildapi.BuildTriggerPolicy, older buildapi.BuildTriggerPolicy, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if trigger.Type != older.Type {
```

Comment:
Please use a deep reflection and exclude the fields you want to copy (see upstream for examples) instead of doing this.  This is extremely fragile - we want a "whitelist" approach for "is this allowed to change, not blacklist.